eng/pipelines/templates/steps/use-rust.yml

```diff
@@ -1,42 +1,100 @@
 parameters:
   - name: Toolchain
     type: string
-    default: stable
+    default: active
+    # Expexted values: 'stable', 'nightly', 'msrv', 'active' or a specific toolchain version
+    # 'msrv' will read the MSRV from azure_core
+    # 'active' will use the active toolchain for the working directory, which will be from rust-toolchain.toml, from a
+    #   folder override or the rustup default toolchain
   - name: MaxAttempts
     type: number
     default: 3
+  - name: WorkingDirectory
+    type: string
+    default: $(System.DefaultWorkingDirectory)
+  - name: SetDefault
+    type: boolean
+    default: true
 
 steps:
-  - pwsh: |
-      $toolchain = '${{ parameters.Toolchain }}'
-      if ($toolchain -eq 'msrv') {
-          Write-Host "Reading MSRV from azure_core"
-          $toolchain = cargo read-manifest --manifest-path ./sdk/core/azure_core/Cargo.toml | ConvertFrom-Json | Select-Object -ExpandProperty rust_version
-      }
+- pwsh: |
+    $Toolchain = '${{ parameters.Toolchain }}'
```

Comment:
We should probably move this inline code into a script

Reply:
We can't as it's needed in a deployment job

Reply:
We could have the deployment job inline its call to `rustup install` and have the general purpose task use a script file